### PR TITLE
add basic test suites for hooks folder

### PR DESCRIPTION
**Pull request by cescoferraro**
This PR creates basic test suites for the hooks placed at the hooks folder

## Changes

- define typing for existing hooks
- add tests

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -39,7 +39,8 @@
   "scripts": {
     "start": "craco start",
     "build": "craco build",
-    "test:unit": "craco test --watchAll=false",
+    "test:work": "craco test --watchAll=false --coverage ./src/hooks --collectCoverageFrom='src/hooks/**/*.(tsx|ts)'",
```
Comment (xoscar) [inline web/package.json:42]: Did this leak into the commit?

File: web/package.json [hunk 2]
```diff
@@ -78,7 +79,8 @@
   "jest": {
     "transformIgnorePatterns": [
       "/node_modules/(?!antd|@ant-design|rc-.+?|@babel/runtime).+(js|jsx)$"
-    ]
+    ],
+    "testMatch": [ "<rootDir>/src/hooks/**/*.test.tsx" ]
```
Comment (xoscar) [inline web/package.json:83]: Shouldn't this match everything not only hooks?

File: web/src/hooks/useDAGChart.ts [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
+import {Dag} from 'd3-dag';
 import _ from 'lodash';
-import { TSpanMap } from '../components/Diagram/components/DAG';
+import {TSpanMap} from '../components/Diagram/components/DAG';
 
-export const useDAGChart = (spanMap: TSpanMap = {}) => {
+export const useDAGChart = (
+  spanMap: TSpanMap = {}
+): void | {
+  dag: Dag<{id: string; parentIds: string[]}, undefined>;
```
Comment (xoscar) [inline web/src/hooks/useDAGChart.ts:9]: Should we encapsulate this into an interface?

File: web/src/hooks/usePolling.ts [hunk 4]
```diff
@@ -6,18 +6,26 @@ interface usePollingProps {
   isPolling: boolean;
 }
 
-const usePolling = ({callback, delay = 1000, isPolling}: usePollingProps) => {
+const usePolling = ({callback, delay = 1000, isPolling}: usePollingProps): void => {
   useEffect(() => {
     let interval: any = null;
 
+    console.log(1);
```
Comment (xoscar) [inline web/src/hooks/usePolling.ts:13]: I guess these logs aren't supposed to be here right?

File: web/src/hooks/usePolling.test.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import {renderHook} from '@testing-library/react-hooks';
+import usePolling from './usePolling';
+
+test('usePolling', async () => {
+  const callback = jest.fn();
+  const payload = {callback, isPolling: true, delay: 0};
+  await renderHook(() => usePolling(payload));
+
+  expect(callback).toBeCalledTimes(0);
```
Comment (xoscar) [inline web/src/hooks/usePolling.test.tsx:9]: Shouldn't the tests be more extensive to cover the different scenarios, like when the `isPolling` could change, etc?

**Review by xoscar — CHANGES_REQUESTED** [review]
Hey @cescoferraro I left some comments. Let me know what you think.

A separate question: Could we move the test files under a `__tests__` folder inside of the same `hooks` one? Thanks!

File: web/src/hooks/usePolling.test.tsx (hunk 5 above)
Comment (cescoferraro) [reply web/src/hooks/usePolling.test.tsx:9]: i am focusing on getting test suites present. 
testing setInterval is a bit tricky.
I am aiming on getting coverage up on those 3 folders.

File: web/src/hooks/useDAGChart.ts (hunk 3 above)
Comment (cescoferraro) [reply web/src/hooks/useDAGChart.ts:9]: sure

File: web/package.json (hunk 2 above)
Comment (cescoferraro) [reply web/package.json:83]: my bad

File: web/package.json (hunk 1 above)
Comment (cescoferraro) [reply web/package.json:42]: my bad

**Review by xoscar — APPROVED** [review]
Looking good! 🚀

**Pull request merged**